tanjun/checks.py

```diff
@@ -297,6 +297,39 @@ async def __call__(
         return self._handle_result((permissions & self.permissions) == self.permissions)
 
 
+class HasAnyRoleCheck(_Check):
+    __slots__ = ("required_roles",)
+
+    def __init__(
+        self,
+        roles: collections.Sequence[typing.Union[hikari.SnowflakeishOr[hikari.Role], str]] = [],
+        *,
+        error_message: typing.Optional[str] = "You do not have the required roles to use this command!",
+        halt_execution: bool = True,
+    ) -> None:
+        super().__init__(error_message, halt_execution)
+        self.required_roles = set(roles)
+
+    async def __call__(self, ctx: tanjun_abc.Context, /) -> bool:
+        if not ctx.member:
+            return self._handle_result(False)
+
+        member_roles = ctx.member.get_roles()
+
+        for member_role in member_roles:
+            if result := self.check_roles(member_role):
+                return self._handle_result(result)
+        return self._handle_result(False)
+
+    def check_roles(self, member_role: hikari.Role) -> bool:
+        for check in self.required_roles:
+            if isinstance(check, int) and member_role.id == check:
+                return True
+            elif isinstance(check, str) and member_role.name == check:
+                return True
```

Comment:
This could probably be simplified to
```suggestion
            if member_role.id == check or member_role.name == check:
                return True
```